esmvaltool/diag_scripts/ocean3d/regions.py

```diff
@@ -0,0 +1,121 @@
+"""APPLICATE/TRR Ocean Diagnostics."""
+import logging
+import os
+import numpy as np
+from scipy.interpolate import interp1d
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def hofm_regions(region, lon2d, lat2d):
+    """Define regions for data selection.
```

Comment:
There is some region functionality in the preprocessor. I believe that the version here might do some things better for 2d coordinates, but in that case it could be a good idea to improve the relevant preprocessor instead?